postgres/database.go

```diff
@@ -35,11 +35,34 @@ func NewDB(ctx context.Context, connString string) (*DB, error) {
 // pass on unsanitised user input!
 func (db *DB) SwitchSchema(ctx context.Context, schema string) error {
 	slog.Info("Switching postgres schema", "schema", schema)
-	if _, err := db.Exec(ctx, fmt.Sprintf("BEGIN; SELECT pg_advisory_xact_lock(1); CREATE SCHEMA IF NOT EXISTS %s; COMMIT;", schema)); err != nil {
-		return fmt.Errorf("cannot create schema '%v': %w", schema, err)
+	_, err := db.Exec(
+		ctx,
+		fmt.Sprintf(
+			"BEGIN; SELECT pg_advisory_xact_lock(1); CREATE SCHEMA IF NOT EXISTS %s; SET search_path TO %s; COMMIT;",
```

Comment:
Ik had die `pg_advisory_xact_lock` toegevoegd om een race conditie te voorkomen als de unit testen en de integration testen op hetzelfde moment hetzelfde schema proberen aanmaken. Als testen nu altijd aparte schema's gaan gebruiken, is dit wss niet meer nodig. Al kan het ook geen kwaad denk ik om het te laten staan.

Reply:
Ah ok, ik wist niet goed waarom je daar een lock nodig zou hebben 😛